### PR TITLE
scala-cli: 0.1.5 -> 0.1.6, add updater script

### PR DESCRIPTION
**Pull request by kubukoz**
###### Description of changes

Adding an updater script to ease/automate future updates. As part of that I extracted the source urls/hashes to a JSON file to make it more declarative/editable by machines. The updater was used to perform the update to 0.1.6.

https://github.com/VirtusLab/scala-cli/releases/tag/v0.1.6

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [x] x86_64-darwin
  - [ ] aarch64-darwin
- [x] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/tools/build-managers/scala-cli/update.nix [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+      --arg platform "$platform" \
+      '{asset: $asset, sha256: $sha256, platform: $platform}')
+    platform_assets+=($asset_object)
+  done
+
+  printf '%s\n' "''${platform_assets[@]}" | \
+    jq -s "map ( { (.platform): . | del(.platform) }) | add" | \
+    jq --arg version $latest_version \
+      '{ version: $version, assets: . }' > $sources_json
+''
```
Comment (turboMaCk) [inline pkgs/development/tools/build-managers/scala-cli/update.nix:37]: do you really want to make this script writable to store? I think maybe using script with nix shebang would be better option for this. For instance this is what we use to update elm https://github.com/NixOS/nixpkgs/blob/master/pkgs/development/compilers/elm/update.sh

Comment (kubukoz) [reply pkgs/development/tools/build-managers/scala-cli/update.nix:37]: I don't mind really, I saw e.g. Sublime's: https://github.com/NixOS/nixpkgs/blob/c10e14acc786808d4bd07aabcbe0adc15af13ac0/pkgs/applications/editors/sublime/4/common.nix#L122-L139

Are there any downsides to this, aside from this being rebuilt after every update?

Comment (turboMaCk) [reply pkgs/development/tools/build-managers/scala-cli/update.nix:37]: well I think running a plain shell script is a bit simpler. also I assume this script is not really useful outside of nixpkgs maintenance. But there are no rules for this as far as I know. You can keep this as is. I would personally create shell script that is using nix-shell myself but ultimately you'll be one working with this not me.

Comment (kubukoz) [reply pkgs/development/tools/build-managers/scala-cli/update.nix:37]: it's fine by me ;)

**Comment by kubukoz** [comment]
is it OK to merge?

**Review by turboMaCk — APPROVED** [review]

**Pull request merged**